### PR TITLE
Fix installer migration

### PR DESCRIPTION
**Pull request by jameskitt616**
- I added a new `generateKey` form to `public/install/forms.php` because the env variable apparently isnt being loaded properly if the key generation and database migration is in one form. And the DB migration needs the `APP_KEY` loaded
- I formatted a few Lines in `public/install/index.php`. But the only new stuff happened from Line 233.

### CODE REVIEW

File: public/install/index.php [hunk 1]
```diff
@@ -64,189 +64,209 @@ function cardStart($title, $subtitle = null): string
...
+                    class="w-1/3 min-w-fit mt-2 px-4 py-2 font-bold rounded-md bg-sky-500 hover:bg-sky-600 shadow-sky-400 focus:outline-2 focus:outline focus:outline-offset-2 focus:outline-sky-500"
+                    name="generateKey">Submit
+                </button>
+            </div>
+        </form>
     <?php
-}
+    }
 
-// DB Migration & APP_KEY Generation
```
Comment (S0ly) [inline public/install/index.php:233]: I dont get why instread of fixing the bug you created a totally new form that is not really needed ?

**Comment by S0ly** [comment]
@jameskitt616

**Comment by jameskitt616** [comment]
@S0ly why did you ping me without a message?

**Review by S0ly — DISMISSED** [review]

**Comment by S0ly** [comment]
I don't know why my GitHub is bugged

**Comment by S0ly** [comment]
I was saying I dont get why instread of fixing the bug you created a totally new form that is not really needed ?

**Comment by jameskitt616** [comment]
I tried "fixing the bug" but apparently when setting the env variables AND trying to access it instantly (in the same form for the migration) it doesnt really work. probably because the state is not updated yet.
I tried to fix it with updating the env variables manually and clearing the settings cache via laravel command during runtime, but it didnt work.
If you got a better idea how to fix it, or know laravel better than i do, i am glad to have some help.
(i also don't think it's to bad to have an extra form for this, but i do agree that it's kinda weird to have)

**Comment by S0ly** [comment]
have you tried to add a sleep for like 3 seconds ?

**Comment by jameskitt616** [comment]
I tried to add a sleep for 1s but that should be plenty of time to set a env variable

**Comment by jameskitt616** [comment]
So what is happening with this now?

**Comment by S0ly** [comment]
you didnt told me if adding a sleep fixed the problem ? is there a more simple less overkill solution than creating a new form ?

**Comment by jameskitt616** [comment]
I already told you at least twice that a sleep of 1s didnt solve it. And 1s should be plenty of time for this.
> is there a more simple less overkill solution than creating a new form ?

yes, creating the `APP_KEY` in the previous step should do the trick as well.
Should i do that?

**Comment by S0ly** [comment]
since checkDB step set all the env value related to DB maybe yes it could be great to merge it whit this step :)

**Comment by jameskitt616** [comment]
Alright will do, maybe i find the time today. I will let you know here when i am done.

**Comment by jameskitt616** [comment]
Updated it so it will generate the `APP_KEY` now in the database setup. Please review the MR.

**Comment by S0ly** [comment]
you tested it and it work ?

**Comment by S0ly** [comment]
my GitHub have the review process that is glitched I dont know why :(

**Comment by jameskitt616** [comment]
Yes tested it for normal case and for a exception case. both worked.

**Comment by S0ly** [comment]
okay :) thanks you

**Pull request merged**